great_expectations/self_check/util.py

```diff
@@ -1930,6 +1931,14 @@ def check_json_test_result(test, result, data_asset=None):
             elif key == "unexpected_list":
                 # check if value can be sorted; if so, sort so arbitrary ordering of results does not cause failure
                 if (isinstance(value, list)) & (len(value) >= 1):
+                    # dictionary handling isn't implemented in great_expectations.core.data_context_key.__lt__
```

Comment:
@NathanFarmer I ❤️ the idea here; I have two questions about it:
1) Would it be better to follow up on your comment and update `great_expectations.core.data_context_key.__lt_` or not?
2) If we must do this "consistent sort" here, is there a strong justification for using `itemgetter` instead of just `lambda`?  If at all possible, I would prefer the `lambda` (for simplicity).

Thank you!  I love how this idea absolves us from the need for the computationally-prohibitive `row_number()`!

Reply:
I clarified the __lt__ comment (it is actually looking at the built-in Python class that is passed). I also changed `itemgetter` to `lambda`.